src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,625 @@
+            System.out.println("I'm sorry, but I don't know what that means :-(");
+        }
+
+        public void showError(String s) {
+            System.out.println(s);
+        }
+
+        public void marked(Task task) {
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.println(task.toString());
+        }
+
+        public void unmarked(Task task) {
+            System.out.println("OK, I've marked this task as not done yet:");
+            System.out.println(task.toString());
+        }
+
+        public void added(Task task) {
+            System.out.println("Got it. I've added this task:");
+            System.out.printf("  added: %s\n", task.toString());
+            System.out.printf("Now you have %s tasks in the list.\n", taskList.count());
+        }
+
+        public void deleted(Task task) {
+            System.out.println("Got it. I've deleted this task:");
+            System.out.printf("  added: %s\n", task.toString());
+            System.out.printf("Now you have %s tasks in the list.\n", taskList.count());
+        }
+
+        public void exit() {
+            System.out.println("Bye!");
+        }
+
+        public void showLine() {
+            System.out.println("_______________________________");
+        }
+    }
+
+    private class TaskList {
+        private ArrayList<Task> all;
+
+        public TaskList() {
+            all = new ArrayList<>();
+        }
+
+        public void displayList() {
+            System.out.println("Here are the tasks in your list:");
+            for (int i = 0; i < count(); i++) {
+                System.out.printf("%s. %s\n", i + 1, all.get(i).toString());
+            }
+        }
+
+        public void findList(String target) {
+            System.out.println("Here are the tasks in your list with the keyword:");
+            for (int i = 0; i < count(); i++) {
+                if (all.get(i).getDescription().indexOf(target) != -1) {
+                    System.out.printf("%s. %s\n", i + 1, all.get(i).toString());
+                }
+            }
+        }
+
+        public void addTask(Task task) {
+            all.add(task);
+            ui.added(task);
+        }
+
+        public void deleteTask(int index) {
+            Task deleted = all.get(index - 1);
+            all.remove(index - 1);
+            ui.deleted(deleted);
+        }
+
+        public Task getTask(int index) {
+            return all.get(index);
+        }
+
+        public int count() {
+            return all.size();
+        }
+    }
+
+    private class Storage {
+        private String filePath;
+
+        public Storage(String path) {
+            filePath = path;
+        }
+
+        /**
+         * Loads data from the text file "Duke.txt". Store the data in the taskList instance
+         *
+         * @throws FileNotFoundException
+         */
+        public void loadData() throws FileNotFoundException {
+            File f = new File(filePath);
```

Comment:
Perhaps you could consider using more descriptive names rather than short form names? I found a number of such instances in the code.